kernel/core/maths/matfuncs.hpp

```diff
@@ -3684,8 +3685,9 @@ static Matrix vectShiftCol(const MatFuncData& funcData, const MatFuncErrorInfo&
 static Matrix vectShiftRow(const MatFuncData& funcData, const MatFuncErrorInfo& errorInfo)
 {
     // Transpose the matrix (including restructuring the memory) to get a row wise representation of the data
-    Matrix _mResult = Matrix(funcData.mat1);
+    Matrix _mResult = funcData.mat1;
     _mResult.transpose();
+    // NOTE: This should not be necessary, but it does not work without it. Maybe the MatFuncData constructor does not copy the transposed state.

```

Comment:
That's weird, have to investigate.

Reply:
@raphael-sys Why do you think that the intended solution is not working? If I comment it out, it works as expected (ignore those `3x3` outputs):
![image](https://github.com/numere-org/NumeRe/assets/87755490/a4b15813-1ffc-450a-b27a-155b5bb1624d)

Reply:
Cant reproduce the error now. So it probably is fine.